pkg/ipam/poolallocator/allocator_test.go

```diff
@@ -135,7 +136,36 @@ func TestAllocateNext(t *testing.T) {
 	stopCh := make(chan struct{})
 	defer close(stopCh)
 
-	poolName := uuid.New().String()
+	poolName := "fakePool"
+	ipRange := crdv1a2.IPRange{
+		Start: "10.2.2.100",
+		End:   "10.2.2.120",
+	}
+	subnetInfo := crdv1a2.SubnetInfo{
+		Gateway:      "10.2.2.1",
+		PrefixLength: 24,
+	}
+	subnetRange := crdv1a2.SubnetIPRange{IPRange: ipRange,
+		SubnetInfo: subnetInfo}
+
+	pool := crdv1a2.IPPool{
+		ObjectMeta: metav1.ObjectMeta{Name: poolName},
+		Spec:       crdv1a2.IPPoolSpec{IPRanges: []crdv1a2.SubnetIPRange{subnetRange}},
+	}
+
+	allocator := newTestIPPoolAllocator(&pool, stopCh)
+	assert.Equal(t, 21, allocator.Total())
+
+	validateAllocationSequence(t, allocator, subnetInfo, []string{"10.2.2.100", "10.2.2.101"})
+}
+
+// This test verifies correct behavior in case of update conflict. Allocation should be retiried
+// Taking into account the latest status
+func TestAllocateConflict(t *testing.T) {
```

Comment:
I don't find how the test simulate a case of update conflict and how it is different from `TestAllocateNext`: there is one line difference about whether check allocator is nil

Reply:
Seems like a merge error, that test was around in prior revisions in allocator_test.go. I'll remove it.